docker/docker/src/main/java/org/arquillian/cube/docker/impl/client/DockerImageController.java

```diff
@@ -0,0 +1,34 @@
+package org.arquillian.cube.docker.impl.client;
+
+import org.arquillian.cube.docker.impl.client.config.CubeContainer;
+import org.arquillian.cube.docker.impl.docker.DockerClientExecutor;
+import org.arquillian.cube.spi.Cube;
+import org.arquillian.cube.spi.CubeRegistry;
+import org.arquillian.cube.spi.event.lifecycle.AfterDestroy;
+import org.jboss.arquillian.core.api.Instance;
+import org.jboss.arquillian.core.api.annotation.Inject;
+import org.jboss.arquillian.core.api.annotation.Observes;
+
+import static org.arquillian.cube.impl.client.CubeLifecycleController.validateAndGet;
+
+
+public class DockerImageController {
+
+    @Inject
+    private Instance<DockerClientExecutor> dockerClientExecutorInstance;
+
+    public void removeImage(@Observes AfterDestroy event, CubeRegistry registry, CubeDockerConfiguration configuration) {
+        if (configuration.isCleanBuildImage()) {
+            Cube cube = validateAndGet(registry, event.getCubeId());
```

Comment:
You don't need  to do a validateAndGet, I would do directly a get and if cube == null then log a warning message and ignore that cube.